examples/simple-usage/lifecycle.ts

```diff
@@ -0,0 +1,23 @@
+import { TaskExecutor } from "@golem-sdk/golem-js";
+
+/**
+ * This example shows how to use the TaskExecutor lifecycle events.
+ */
+(async function main() {
+  console.log("Creating TaskExecutor and registering for lifecycle events...");
+  const executor = await TaskExecutor.create("golem/alpine:latest");
+
+  executor.events.on("terminating", () => {
```

Comment:
I don't think this event adds any value except for showcasing the mechanism. Developers are used to events which represent something that something "happened", not "is happening, or is about to happen". Please remove the `terminating` event.